src/components/charts/BarChart.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react"
+import merge from "deepmerge"
+
+import {
+  ChartWrapper,
+  RecordSet,
+  RecordEntry,
+  ChartProps,
+} from "./ChartWrapper"
+
+interface BarChartProps extends ChartProps {
+  /**
+   * The order in which the x or y axis labels should be displayed.
+   */
+  xOrder?: string[]
+  /**
+   * WHether the bar chart should be horizontal.
+   */
+  horizontal?: boolean
+  /**
+   * Whether the bar chart should be stacked.
+   */
+  stacked?: boolean
+  /**
+   * The minimum value of the x (horizontal) or y axis.
+   */
+  min?: number
+  /**
+   * The maximum value of the x (horizontal) or y axis. Useful when data labels overflow.
+   */
+  max?: number
+  /**
+   * The width of the chart object. Defaults to 100% so it is responsive.
+   */
+  showLabels?: boolean
+}
+
+interface BarSeriesEntry {
+  name: string
+  data: (string | number)[]
+}
+
+export type BarSeries = BarSeriesEntry[]
+
+const makeDataLabelOptions = (
+  stacked: boolean,
+  horizontal: boolean,
+  min: number,
+  max: number
+) => {
+  const labelOptions: Record<string, unknown> = {}
+  if (stacked) {
+    labelOptions["dataLabels"] = {
+      textAnchor: "middle",
+      style: {
+        colors: ["#FFF"],
+      },
+    }
+  } else {
+    if (horizontal) {
+      labelOptions["dataLabels"] = {
+        textAnchor: "left",
+        offsetX: 20,
+        style: {
+          colors: ["#222"],
+        },
+      }
+    } else {
+      labelOptions["dataLabels"] = {
+        offsetY: -20,
+        style: {
+          colors: ["#222"],
+        },
+      }
+    }
+  }
+
+  if (horizontal) {
+    labelOptions["xaxis"] = {
+      min,
+      max,
+      axisTicks: {
+        show: false
+      },
+      tickAmount: 5
+    }
+  } else {
+    labelOptions["yaxis"] = {
+      min,
+      max,
+    }
+  }
+  return labelOptions
+}
+
+const sortFieldByArrayIndex = (
+  arr: RecordSet,
+  order: unknown[],
+  field: string
+): RecordSet => {
+  const newArr = [...arr]
+  newArr.sort(
+    (a: RecordEntry, b: RecordEntry) =>
+      order.indexOf(a[field]) - order.indexOf(b[field])
+  )
+  return newArr
+}
+
+const makeSeries = (data: RecordSet, y: string[]): BarSeries => {
+  const series = y.map(label => ({
+    name: label,
+    data: data.map(entry => entry[label]),
+  }))
+
+  return series
+}
+
+export const BarChart: React.FC<BarChartProps> = ({
+  data,
+  x,
+  y,
+  colors,
+  xOrder,
+  horizontal = false,
+  stacked = false,
+  title,
+  subtitle,
+  options = {},
+  min = 0,
+  max = undefined,
+  width = "100%",
+  height = "600px",
+  showLabels = false,
+}: BarChartProps) => {
+  const sortedData = xOrder ? sortFieldByArrayIndex([...data], xOrder, x) : data
+
+  const series = makeSeries(sortedData, y)
+
+  const labels = data.map(entry => entry[x])
+
+  const defaultOptions = {
```

Comment:
Just a note that the name `defaultOptions` has the connotation that this is what you get if you don't specify anything. Perhaps `baseOptions` or `chartOptions` ?

Reply:
Good point. Will do